src/commands/register.ts

```diff
@@ -118,40 +127,61 @@ export default class ComponentRegister extends Command {
     const project_name = `register.${resourceRefToNodeRef(component_spec.name)}.${tag}`;
     const compose_file = DockerComposeUtils.buildComposeFilepath(this.app.config.getConfigDir(), project_name);
 
-    await DockerComposeUtils.writeCompose(compose_file, yaml.dump(compose));
+    await DockerBuildXUtils.writeCompose(compose_file, yaml.dump(compose));
 
     let build_args: string[] = [];
     for (const service_config of Object.values(component_spec.services || {})) {
       build_args = build_args.concat((await this.getBuildArgs(service_config)).map(arg => {
         return `${arg}`;
       }));
     }
+
     build_args = build_args.filter((value, index, self) => {
       return self.indexOf(value) === index;
     }).reduce((arr, value) => {
-      arr.push('--build-arg');
-      arr.push(value);
+      arr.push("--set");
+      arr.push(`*.args.${value}`);
       return arr;
     }, [] as string[]);
 
     try {
-      await DockerComposeUtils.dockerCompose(['-f', compose_file, 'build', ...build_args], {
-        stdio: 'inherit',
+      const is_local = this.app.config.api_host.includes("localhost");
```

Comment:
Lets move all of this logic into the BuildX utils. You should just have something like.
```
await DockerBuildXUtils.createBuilder();
```